src/main/java/cceclipseplugin/editor/listeners/DirectoryListener.java

```diff
@@ -36,24 +38,33 @@
 	protected boolean handleProject(IResourceDelta delta) {
 		IProject p = (IProject) delta.getResource();
 		ProjectMetadata projectMeta = PluginManager.getInstance().getMetadataManager().getProjectMetadata(p.getFullPath().toString());
-		RequestManager rm = PluginManager.getInstance().getRequestManager();
+		PluginManager pm = PluginManager.getInstance();
+		RequestManager rm = pm.getRequestManager();
 		
 		if (delta.getKind() == IResourceDelta.REMOVED) {			
 			// Project was renamed
 			if ((delta.getFlags() & IResourceDelta.MOVED_TO) != 0) {
-				String newName = delta.getMovedToPath().lastSegment();
-				String newPath = delta.getMovedToPath().toString();
-				
-				rm.renameProject(projectMeta.getProjectID(), newName, newPath);
-				System.out.println("sent project rename request: renamed to \"" + newName + "\"; path changed to : " + newPath);
-				return false;
+				if (pm.isProjectInWarnList(p.getName(), ProjectRenameNotification.class)) {
+					pm.removeProjectFromWarnList(p.getName(), ProjectRenameNotification.class);
+				} else {
+					String newName = delta.getMovedToPath().lastSegment();
+					String newPath = delta.getMovedToPath().toString();
+					
+					rm.renameProject(projectMeta.getProjectID(), newName, newPath);
+					System.out.println("sent project rename request: renamed to \"" + newName + "\"; path changed to : " + newPath);
+					return false;
+				}
+			} else if ((delta.getFlags() & IResourceDelta.REMOVED) != 0) {
+				if (pm.isProjectInWarnList(p.getName(), ProjectDeleteNotification.class)) {
+					pm.removeProjectFromWarnList(p.getName(), ProjectDeleteNotification.class);
+				} else {
+					System.out.println("deleting project");
+					// Project was deleted from disk
+					System.out.println("unsubscribed from project due to removal from disk");
+					PluginManager.getInstance().getRequestManager().unsubscribeFromProject(projectMeta.getProjectID());
```

Comment:
I'm thinking that this would probably be a good idea since most other directory changes (file creation, deletion, renaming, etc) are reflected on the server